src/App.tsx

```diff
@@ -1,19 +1,181 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type ToDo = {
+  userId: number,
+  id: number,
+  title: string,
+  completed?: boolean,
 };
 
+type State = {
+  todos: ToDo[],
+  id: number,
+  title: string,
+  userId: number,
+  invalidTitle: boolean,
+  invalidUser: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    todos: [...todos],
+    id: todos.length + 1,
+    title: '',
+    userId: 0,
+    invalidTitle: false,
+    invalidUser: false,
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!this.state.title || !users.find(person => person.id === this.state.userId)) {
+      this.showErrors();
```

Comment:
І тоді непотрібно буде писати else
```suggestion
      return this.showErrors();
```